examples/chat/weaver_gen.go

```diff
@@ -997,6 +1009,71 @@ func (s sQLStore_server_stub) getImage(ctx context.Context, args []byte) (res []
 	return enc.Data(), nil
 }
 
+// Reflect stub implementations.
+
+type imageScaler_reflect_stub struct {
+	caller func(string, context.Context, []any) ([]any, error)
+}
+
+// Check that imageScaler_reflect_stub implements the ImageScaler interface.
+var _ ImageScaler = (*imageScaler_reflect_stub)(nil)
+
+func (s imageScaler_reflect_stub) Scale(ctx context.Context, a0 []byte, a1 int, a2 int) (r0 []byte, err error) {
+	results, err := s.caller("Scale", ctx, []any{a0, a1, a2})
+	return results[0].([]byte), err
+}
+
+type localCache_reflect_stub struct {
+	caller func(string, context.Context, []any) ([]any, error)
+}
+
+// Check that localCache_reflect_stub implements the LocalCache interface.
+var _ LocalCache = (*localCache_reflect_stub)(nil)
+
+func (s localCache_reflect_stub) Get(ctx context.Context, a0 string) (r0 string, err error) {
+	results, err := s.caller("Get", ctx, []any{a0})
+	return results[0].(string), err
```

Comment:
Do we need to avoid touching results when err is non-nil?

Reply:
I was thinking we should still preserve the return values even if err is non-nil in case a method returns meaningful values alongside a non-nil error? I think [Buffer.ReadBytes](https://pkg.go.dev/bytes#Buffer.ReadBytes) does this, for example.

Reply:
Ah, sorry, you're saying that if err is non-nil, the returned `results` could be an empty slice for example, in which case indexing into it would panic? I overlooked that. I can update the comments to say that that `caller` should always return a valid `results`?

Reply:
Yes, I was worried about: (1) results being too short (2) entries in results not having the right type.

I worry that just saying call should always return a valid results may be easy to miss. What code are we calling: if it is generated code, it might be fine to place such a restriction. If it is hand-written code, I would worry about that.

Reply:
Great point. It is hand-written. I could generate some code to panic if `results` is not well-formed? For example: 

```go
func (s localCache_reflect_stub) Get(ctx context.Context, a0 string) (r0 string, err error) {
    results, err := s.caller("Get", ctx, []any{a0})
    if len(results) != 1 {
        panic(fmt.Errorf("..."))
    }
    if _, ok := results[0].(string); !ok {
        panic(fmt.Errorf("..."))
    }
    return results[0].(string), err
}
```

Reply:
What if we made s.caller responsible for filling the return variables:

```
err = s.caller("Get", ctx, []any{a0}, []any{&r0})
return
```

That will keep the generated code size short. This stub will become a bit more expensive, but that is probably fine for the intended use.

Reply:
Great idea! I switched to your suggestion.